pkg/awsutils/awsutils.go

```diff
@@ -727,6 +730,19 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 // AllocENI creates an ENI and attaches it to the instance
 // returns: newly created ENI ID
 func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string) (string, error) {
+	var sgIds []string
+
+	for _, sgId := range sg {
```

Comment:
Why do we need to do this? I see we're converting them back to []*string in `ValidateSecurityGroups`?

Reply:
My mistake. Thanks for spotting. Somewhere between refactoring, I had this unnecessary double conversion. I will remove it and it will make it much better.